### ISSUE TITLE
Description Design: Responses pile up with Step interaction

### ISSUE THREAD

**Issue by terracoda** [phetsims/molecules-and-light#285]
In VO responses pile up when pressing and holding Step button.
Not sure of solution.

Perhaps @Matthew-Moore240 and @jessegreenberg could explore?

**Comment by jessegreenberg** [comment]
Over slack @terracoda recommended that we change the design to only alert the first vibration alert "...molecule stretching." rather than alert that both "..molecule stretching" AND "...molecule contracting".

This will reduce verbosity quite a bit.

I will also think about ways to manage the utterances better.

**Comment by jessegreenberg** [comment]
I think this is improved with the above commit, @terracoda and @Matthew-Moore240 can you please review?

https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-dev.30/phet/molecules-and-light_en_phet.html

**Comment by terracoda** [comment]
@jessegreenberg, this works really nicely. I will use it for the interview today.

**Comment by terracoda** [comment]
You can close this issue if there is no more work to do on this until I get feedback.

@Matthew-Moore240, the wording in the response is different. It uses "contracting" instead of "stretching". 

Is that intentional?

**Comment by jessegreenberg** [comment]
>  It uses "contracting" instead of "stretching".

This is because it is describing what the molecule does when it first absorbs the photon.  If the molecule bonds immediately start contracting it will say "contracting" instead of stretching. We could change things if it is better for alerts to always say "stretching".

**Comment by terracoda** [comment]
@Matthew-Moore240, do you have any thoughts? Should it say "contracting and stretching"?

**Comment by terracoda** [comment]
@Matthew-Moore240, please review the responses currently being offered up in the Step Forward mode and see if you think they need to be combined or changed.

I have no data from blind users on the Step Forward button yet, but the responses could still be really helpful for other learners. 

The slow motion, "Slow" radio button seems to be most useful for blind users, and so I think efforts on optimizing these phrasing for these 2 modes could be useful.

**Comment by Matthew-Moore240** [comment]
I think if we only want one version of the response either "stretching" or "contracting"  we should go with "stretching". Just so I understand we would only want 

**For stepping through**
_{{Light source}} photon {{absorbed}} and bonds of {{molecule}} molecule {{stretching}}_

**For normal speed**
_Bonds of molecule stretching._
**After first response**
_Stretching_

**Comment by terracoda** [comment]
@Matthew-Moore240, I am not opposed to using both words:
{{Light source}} photon {{absorbed}} and bonds of {{molecule}} molecule {{ contracting and stretching}}

And we could make the strings briefer in a few ways:
- Replace "and" with a comma, 
- use consistent tenses: simple present for absorbed and emitted, and simple present for reactions or non-reactions. The present progressive doesn't pair well with the simple past
- And in you could short "breaks part into" to "breaks into" (not spelling there is a typo "part" should be "apart")
- could consider not repeating molecule in the emission phrase

- {{Light source}} photon {{absorbed}}, bonds of {{molecule-name}} {{contract and stretch}}.
- Absorbed {{light source}} photon {{emitted}} {{direction-of-emission}}.

**Examples:**
- Infrared photon absorbed, bonds of Carbon Monoxide molecule contract and stretch.
- Infrared photon absorbed, bonds of Carbon Dioxide molecule bend up and down.
- Microwave photon absorbed, Water molecule rotates clockwise.
- Microwave photon absorbed, Water molecule rotates counter-clockwise.
- Microwave photon passes through Methane molecule. (currently this uses plural "photons", not sure why)
- Ultraviolet photon passes through Nitrogen Dioxide molecule.
- Ultraviolet photon absorbed, Nitrogen Dioxide molecule breaks into NO and O.
- Ultraviolet photon absorbed, Ozone molecule breaks into O2 and O.

Also, for emission, need to stick with same tenses, emitted, rather than emitting:
- Absorbed Infrared photon emitted from Methane molecule up and to the right.

And if you think it is safe to remove the molecule string emissions would look like this:
-  Absorbed Infrared photon emitted up and to the right.
- Absorbed Microwave photon emitted left.

**Actually, @Matthew-Moore240,** I think it would be good to review the phrases currently being used in slow motion with the phrases being used in Step Forward and make sure they are consistent in phrasing and only change if there is a good reason for them to be different.

I think we changed the slow motion ones, but have re-looked at the Step Forward ones.

**Comment by terracoda** [comment]
And for the record, I like the present progressive ("ing") at Normal speed. I am not convinced it works in Slow motion or Step Forward, though when combined with other verb tenses.

**Comment by terracoda** [comment]
@jessegreenberg, @Matthew-Moore240 and I worked on some tweaks/rephrasing of the current normal speed, slow speed and step forward context responses.

We tried to align the verb tenses, shorten the slow motion responses to account for slower screen reader speeds and align the phrasing a little better across the three speed modes.

Examples are found in the [Table: Review of Molecule Responses](https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#heading=h.8gkq5oll4829) in the design document.

Marking for discussion at a11y design meeting.

**Comment by terracoda** [comment]
@jessegreenberg, we didn't get to this item in Tuesday's meeting.
Could you review the  [Table: Review of Molecule Responses](https://docs.google.com/document/d/1r9D3vq4QvmmXkQ5G6HexpybUv_vL6LrYFb3wgPfUdWg/edit#heading=h.8gkq5oll4829), and let us know if all the suggested changes are possible?

The main changes are in Slow motion and in Step Forward modes, though there are a couple of tweaks to the phrases in Normal as well.

For Slow, we now have a verbose and brief pattern very similar to Normal. This is to account for screen readers set to slower speech rates, and we also think it streamlines the design and experience.

For Step forward there are list of changes...please let us know if you have any questions.

**Comment by terracoda** [comment]
@jessegreenberg, as I noted on slack. 

>  I just finished the examples in the molecule table. I didn't realize all the examples were not filled in.

**Comment by jessegreenberg** [comment]
These alerts are complete and are ready for review at https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-dev.46/phet/molecules-and-light_a11y_view.html

**Comment by jessegreenberg** [comment]
Note that the original issue was that alerts were piling up. But what is left to review is the new alert content.

**Comment by terracoda** [comment]
- [x] verified Step Forward phrases in A11y View
- [ ] verified Step Forward phrases with VoiceOver

**Comment by terracoda** [comment]
@jessegreenberg, I am not experiencing any pile ups with the step forward button.
All the detailed Step Forward response look great.

There is one spot where things get awkward which is related to what I noticed in issue for the dynamic descriptions - once Ozone or Nitrogen Dioxide break apart, their pieces float away. Once they are off screen we need one more phrase (Phase 4) to fire. This will make it clear the pieces are gone and that learner needs to reset or change molecule. The same phrase should work.

- Molecule pieces gone. Reset or change molecule.

**Comment by jessegreenberg** [comment]
> Molecule pieces gone. Reset or change molecule.

This has been added in https://github.com/phetsims/molecules-and-light/issues/323 so I think this issue can be closed.
